working-docs/transient.md

```diff
@@ -0,0 +1,379 @@
+
+### CWE-A: Processor Event Causes Transient Execution
+
+#### Description
+
+A processor event may allow subsequent operations to execute transiently
+(the operations execute without committing to architectural state).
+
+#### Extended Description
+
+Many commodity processors execute operations (such as instructions,
+etc.) out-of-order. For example, a processor may fetch the sequence of
+instructions `A;B;C`, but then execute them in a different order, such as
+`A;C;B`. This out-of-order execution sequence may produce the same result
+as an in-order sequence if the input to `C` does not depend on the output
+of `B`. Although operations on these processors may execute out-of-order,
+they commit in-order to the processor’s architectural state. Hence, all
+operations appear to software as though they had been executed in-order
+by the processor.
+
+Some processor events can cause an out-of-order processor to flush its
+pipeline, discarding the results of operations that have already
+executed but not committed to the processor’s architectural state. For
+example, suppose that while executing `B` in the out-of-order sequence
+`A;C;B`, the processor encounters an event that causes a pipeline flush.
+If `C` executed before `B`, the event may require `C`’s results to be
+discarded during the flush.
+
+When operations execute but do not commit to the processor’s
+architectural state, this is commonly referred to as *transient
+execution*. Operations that execute transiently may have
+microarchitectural side effects that can be detected using timing or
+power analysis techniques. These techniques may allow an attacker to
+infer information about the operations that executed transiently. For
+example, the attacker may be able to infer program data that was
+accessed or used by those operations.
+
+### CWE-B: Transient Data Forwarding from an Operation that Triggers a Processor Event
+
+#### Description
+
+A processor event (for example, a fault or microcode assist) may allow
+incorrect data to be forwarded from the operation that triggered the
+event to operations that execute transiently.
+
+#### Extended Description
+
+Commodity processors may require assistance from microcode (or other
+processor design techniques) to execute certain operations. For example,
+processors that implement virtual memory may use microcode to perform
+virtual-to-physical address translation, check for faults, etc. Prior to
+or concurrent with the microcode assist, some processors may allow
+incorrect data to be forwarded from the operation that triggered the
+assist to operations that execute transiently and are flushed by the
+time the assist completes. These transient operations can affect
+observable microarchitectural state in a manner that could allow an
+attacker to infer program data, such as the incorrect data forwarded by
+the operation that triggered the assist.
+
+### CWE-C: Transient Execution Influenced by Shared Microarchitectural Predictor State
```

Comment:
Gananand: This one seems very close to the issue already. Perhaps this could be “Improper Isolation of Hardware Domains through Shared Micro-architectural State and Transient Execution”. How would this differ from CWE-1189: Improper Isolation of Shared Resources on a System-On-Chip?